runtime/src/test/java/org/eclipse/xpanse/runtime/ServiceConfigurationApiTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.runtime;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import jakarta.transaction.Transactional;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.api.controllers.ServiceConfigurationApi;
+import org.eclipse.xpanse.modules.deployment.ServiceConfigurationManager;
+import org.eclipse.xpanse.modules.models.service.order.ServiceOrder;
+import org.eclipse.xpanse.modules.models.serviceconfiguration.ServiceConfigurationUpdate;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Test for ServiceConfigurationApi.
+ */
+@Slf4j
+@Transactional
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(properties = {"spring.profiles.active=oauth,zitadel,zitadel-testbed"})
+@AutoConfigureMockMvc
+public class ServiceConfigurationApiTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Mock
+    private ServiceConfigurationManager serviceConfigurationManager;
+
+    @InjectMocks
+    private ServiceConfigurationApi serviceConfigurationApi;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        this.mockMvc = MockMvcBuilders.standaloneSetup(serviceConfigurationApi).build();
+    }
+
+    @Test
+    @WithMockUser(roles = {"USER"})
+    void testChangeServiceConfiguration_Success() throws Exception {
```

Comment:
would it be possible to add a full end to end test for this?
1. Register a kafka template. 
2. Request config change. 
3. Validate entries in config table - check that exact number of expected requests are created. 
4. validate the request properties are as expected. 

Then we know that if anything changes in the logic, this test will fail.